src/components/views/login/ServerConfig.js

```diff
@@ -29,8 +29,10 @@ module.exports = React.createClass({
     propTypes: {
         onHsUrlChanged: React.PropTypes.func,
         onIsUrlChanged: React.PropTypes.func,
-        defaultHsUrl: React.PropTypes.string,
-        defaultIsUrl: React.PropTypes.string,
+        initialHsUrl: React.PropTypes.string, // whatever the current value is when we create the component
+        initialIsUrl: React.PropTypes.string, // whatever the current value is when we create the component
+        defaultHsUrl: React.PropTypes.string, // e.g. https://matrix.org
+        defaultIsUrl: React.PropTypes.string, // e.g. https://vector.im
```

Comment:
I'm not convinced that this is absolutely required (and if it is, the prop names are misleading). Before this patch, the "default" URLs were the initial values which populated the input fields. Now, the "initial" URLs are the initial values which populate the input fields, with "default" URLs now indicating what the placeholder text should be. Except this isn't completely true in cases where you modify the input field such that `onXXXChanged` fires in which case the default URL will be used if the one they entered is empty. We also seemingly use the defaults, overriding the explicit value in the input field, if we untick the "show advanced options" checkbox. This smells like bad UX to me as I don't think it is in any way clear than unticking that box will clobber your URLs.

I need you to clarify your intent on these props a bit more than you have done please. I'm also questioning the underlying UX which instigated these changes.